docs/guide/src/example-plugin/plugin.go

```diff
@@ -6,55 +6,115 @@ import (
 	"github.com/tendermint/go-wire"
 )
 
+//-----------------------------------------
+// Structs
+//   * Note the fields in each struct may be expanded/modified
+
+// Plugin State Struct
+//   * Intended to store the current state of the plugin
+//     * This example contains a field which holds the execution count
+//   * Used by go-wire as the encoding/decoding struct to hold the plugin state
+//   * All fields must be exposed (for go-wire)
+//   * The state is stored within the KVStore using the key retrieved
+//     from the ExamplePlugin.StateKey() function
 type ExamplePluginState struct {
 	Counter int
 }
 
+// Transaction Struct
+//   * Stores transaction-specific plugin-customized information
+//     * This example contains a dummy field 'Valid' intended to specify
+//       if the transaction is a valid and should proceed
+//   * Used by go-wire as the encoding/decoding struct to pass transaction
```

Comment:
Again, I'm not thrilled with this `Used by go-wire as the encoding/decoding struct` language